### PR TITLE
Sync CardActions via Firebase

### PR DESCRIPTION
**Pull request by kuttnig**

### CODE REVIEW

**Comment by raphaelseher** [comment]
|File|Coverage [79.85%]|:green_apple:|
|:-|:-:|:-:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/72bfcb6f3f87b9de0eea1c3a3c099f3849861e9f/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|79.85%|:green_apple:|

|Total Project Coverage|24%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
|File|Coverage [49.98%]|:green_apple:|
|:-|:-:|:-:|
|[RowStatus.java](https://github.com/SE2-Gwent/Gwent/blob/787664fc28a6d159af57cf19b833d089ee818adf/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRowStatus.java)|100%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/787664fc28a6d159af57cf19b833d089ee818adf/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|50.21%|:green_apple:|
|[GameFieldRows.java](https://github.com/SE2-Gwent/Gwent/blob/787664fc28a6d159af57cf19b833d089ee818adf/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameFieldRows.java)|41.5%|:green_apple:|
|[Row.java](https://github.com/SE2-Gwent/Gwent/blob/787664fc28a6d159af57cf19b833d089ee818adf/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRow.java)|32.73%|:green_apple:|

|Total Project Coverage|22.79%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
|File|Coverage [48.78%]|:green_apple:|
|:-|:-:|:-:|
|[RowStatus.java](https://github.com/SE2-Gwent/Gwent/blob/ea4246e9e65de14b5871f880e4c2dc20798e4bcb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRowStatus.java)|100%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/ea4246e9e65de14b5871f880e4c2dc20798e4bcb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|48.87%|:green_apple:|
|[GameFieldRows.java](https://github.com/SE2-Gwent/Gwent/blob/ea4246e9e65de14b5871f880e4c2dc20798e4bcb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameFieldRows.java)|41.5%|:green_apple:|
|[Row.java](https://github.com/SE2-Gwent/Gwent/blob/ea4246e9e65de14b5871f880e4c2dc20798e4bcb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRow.java)|32.73%|:green_apple:|

|Total Project Coverage|21.58%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
|File|Coverage [48.82%]|:green_apple:|
|:-|:-:|:-:|
|[RowStatus.java](https://github.com/SE2-Gwent/Gwent/blob/ea4ebad2da95ab8c469e4cb53a46eebc482b605d/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRowStatus.java)|100%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/ea4ebad2da95ab8c469e4cb53a46eebc482b605d/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|48.92%|:green_apple:|
|[GameFieldRows.java](https://github.com/SE2-Gwent/Gwent/blob/ea4ebad2da95ab8c469e4cb53a46eebc482b605d/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameFieldRows.java)|41.5%|:green_apple:|
|[Row.java](https://github.com/SE2-Gwent/Gwent/blob/ea4ebad2da95ab8c469e4cb53a46eebc482b605d/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRow.java)|32.73%|:green_apple:|

|Total Project Coverage|21.59%|:green_apple:|
|:-|:-:|:-:|

**Review by raphaelseher — CHANGES_REQUESTED** [review]

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/GameFieldRows.java [hunk 1]
```diff
@@ -22,7 +49,7 @@ public HashMap<String, Card> meleeRowFor(InitialPlayer player) {
     }
   }
 
-  public HashMap<String, Card> rangedRowFor(InitialPlayer player) {
+  public Row getRangedRowFor(InitialPlayer player) {
```
[the remark this answers is not in the document]
Comment (kuttnig) [reply gamelogic/src/main/java/at/aau/se2/gamelogic/models/GameFieldRows.java:52]: Hab die Getter umgeschrieben. Sollte jetzt wieder funktionieren.

File: app/src/main/java/at/aau/se2/gwent/views/board/BoardFragment.java [hunk 2]
```diff
@@ -150,7 +150,7 @@ private void updateUI(BoardViewData viewData) {
 
     if (viewModel.getOldViewData() != null
         && viewModel.getOldViewData().getSelectedCardId() != null) {
-      playersHandCardViews.get(viewModel.getOldViewData().getSelectedCardId()).setSelected(false);
+      // playersHandCardViews.get(viewModel.getOldViewData().getSelectedCardId()).setSelected(false);
```
Comment (raphaelseher) [inline app/src/main/java/at/aau/se2/gwent/views/board/BoardFragment.java]: Wenn die Zeile einfach auskommentierst, weils crashed, dann kann man mehrere Karten auswählen beim ausspielen.

image

**Comment by raphaelseher** [comment]
|File|Coverage [39.16%]|:green_apple:|
|:-|:-:|:-:|
|[RowStatus.java](https://github.com/SE2-Gwent/Gwent/blob/52dc057ee73a8c4047656e5a5069bc1d82450f6a/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRowStatus.java)|100%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/52dc057ee73a8c4047656e5a5069bc1d82450f6a/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|53.31%|:green_apple:|
|[GameFieldRows.java](https://github.com/SE2-Gwent/Gwent/blob/52dc057ee73a8c4047656e5a5069bc1d82450f6a/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameFieldRows.java)|48.43%|:green_apple:|
|[Row.java](https://github.com/SE2-Gwent/Gwent/blob/52dc057ee73a8c4047656e5a5069bc1d82450f6a/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRow.java)|38.18%|:green_apple:|
|[BoardViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/52dc057ee73a8c4047656e5a5069bc1d82450f6a/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewModel.java)|0%|:green_apple:|
|[BoardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/52dc057ee73a8c4047656e5a5069bc1d82450f6a/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardFragment.java)|0%|:green_apple:|

|Total Project Coverage|22.81%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
Nächstes mal bitte Tests fixen vorm weitergeben. Sonst top!

**Review by raphaelseher — APPROVED** [review]

**Pull request merged**